### PR TITLE
Move FolderWithAllItems request to content SDK

### PR DESCRIPTION
**Pull request by varungu**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @varungu has signed the CLA. Thanks for the pull request!

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java [hunk 1]
```diff
@@ -638,4 +640,92 @@ public BoxIteratorItems sendForCachedResult() throws BoxException {
...
+                    @Override
+                    protected void onSendCompleted(BoxResponse<BoxListItems> response) throws BoxException {
+                        // Do nothing as we don't want this request to be cached
+                    }
+                }.setFields(BoxFolder.ALL_FIELDS)
+                        .setOffset(offset)
+                        .setLimit(limit);
+                batchRequest.addRequest(folderItemsReq);
+            }
+            // TODO: Baymax - Batch Requests should be run in parallel
```
Comment (seema-at-box) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java]: Shouldn't checkin baymax todos on sdk

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java [hunk 2]
```diff
@@ -638,4 +640,92 @@ public BoxIteratorItems sendForCachedResult() throws BoxException {
...
+
+            BoxRequestBatch batchRequest = new BoxRequestBatch().setExecutor(SdkUtils.createDefaultThreadPoolExecutor(10, 10, 3600, TimeUnit.SECONDS));
+            BoxListItems boxListItems = folder.getItemCollection();
+            int offset = boxListItems.offset().intValue();
+            int limit = boxListItems.limit().intValue();
+            while (offset + limit < boxListItems.fullSize()) {
+                offset += limit;
+                limit = 1000;
+
+                BoxRequestsFolder.GetFolderItems folderItemsReq = new BoxRequestsFolder.GetFolderItems(mFolderId, mRequestUrlString + "/items", mSession) {
```
Comment (seema-at-box) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java]: Is there a static string we can use?

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java [hunk 3]
```diff
@@ -638,4 +640,92 @@ public BoxIteratorItems sendForCachedResult() throws BoxException {
...
+            }.setFields(BoxFolder.ALL_FIELDS).setLimit(1000);
+            BoxFolder folder = folderInfoReq.send();
+
+            BoxRequestBatch batchRequest = new BoxRequestBatch().setExecutor(SdkUtils.createDefaultThreadPoolExecutor(10, 10, 3600, TimeUnit.SECONDS));
+            BoxListItems boxListItems = folder.getItemCollection();
+            int offset = boxListItems.offset().intValue();
+            int limit = boxListItems.limit().intValue();
+            while (offset + limit < boxListItems.fullSize()) {
+                offset += limit;
+                limit = 1000;
```
Comment (seema-at-box) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java]: put in a constant

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java [hunk 4]
```diff
@@ -638,4 +640,92 @@ public BoxIteratorItems sendForCachedResult() throws BoxException {
...
+    /**
+    * Get full folder including all information of all children
+    */
+    public static class GetFolderWithAllItems extends BoxRequestItem<BoxFolder, GetFolderWithAllItems> implements BoxCacheableRequest<BoxFolder> {
+        public static final int NO_LIMIT = -1;
+        private static final long serialVersionUID = -146995041590363404L;
+        private String mFolderId;
+        //Limit results count for local fetches
+        private int mLocalLimit = NO_LIMIT;
+
```
Comment (seema-at-box) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java]: not sure if local limit should be in the sdk

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java [hunk 5]
```diff
@@ -638,4 +640,92 @@ public BoxIteratorItems sendForCachedResult() throws BoxException {
...
+        }
+
+        @Override
+        public BoxFolder onSend() throws BoxException {
+            BoxRequestsFolder.GetFolderInfo folderInfoReq = new BoxRequestsFolder.GetFolderInfo(mFolderId, mRequestUrlString, mSession) {
+                @Override
+                protected void onSendCompleted(BoxResponse<BoxFolder> response) throws BoxException {
+                    // Do nothing as we don't want this request to be cached
+                }
+            }.setFields(BoxFolder.ALL_FIELDS).setLimit(1000);
```
Comment (doncung) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java]: Can we get this from the original request.

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java (hunk 2 above)
Comment (varungu) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java]: It is in BoxApiFolder. Can't use that here. I'll look at it separately.

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java (hunk 1 above)
Comment (varungu) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java]: Done.

File: box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java (hunk 4 above)
Comment (varungu) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFolder.java]: Done.

**Comment by doncung** [comment]
+1

**Pull request merged**